src/l3.c

```diff
@@ -343,6 +373,48 @@ l3_log_mmap(const char *msg, const uint64_t arg1, const uint64_t arg2,
     l3_log->slots[idx].arg2 = arg2;
 }
 
+/**
+ * l3_log_write() - 'C' interface to log L3 log-entries using write()
+ */
+void
+l3_log_write(const char *msg, const uint64_t arg1, const uint64_t arg2)
+{
+    L3_ENTRY le     = {0};
+    L3_ENTRY *lep   = &le;
+
+    lep->tid = l3_mytid();
+
+    lep->msg = msg;
+    lep->arg1 = arg1;
+    lep->arg2 = arg2;
+
+    if (write(l3_log_fd, lep, sizeof(*lep)) != sizeof(*lep)) {
+        fprintf(stderr, "%s(): write() failed. errno=%d\n",
+                __func__, errno);
+        return;
+    }
+    return;
+}
+
+/**
+ * l3_log_write_msg() - 'C' interface to log L3 log-entries using write()
+ * Similar to l3_log_write(), but the msg is sprintf()'ed using 'msgfmt' format
+ * specifiers, requiring 2 arguments.
+ */
+void
+l3_log_write_msg(const char *msgfmt, const uint64_t arg1, const uint64_t arg2)
```

Comment:
This is really a temp fn, added for perf-benchmarking - to figure out the overheads of a simple `write()` system call v/s the overhead of doing a `write()` on a msg-buffer where the print format specifiers have been replaced by the argument.

(This function should have gone into the 2nd commit, but got stuck in here. Eventually, this may come out altogether.)